src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: position.x || 0,
+      y: position.y || 0,
+    };
 
-class DeliveryDrone {}
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(steps = 1) {
+    this.coords.y += steps;
+  }
+
+  goBack(steps = 1) {
+    this.coords.y -= steps;
+  }
+
+  goRight(steps = 1) {
+    this.coords.x += steps;
+  }
+
+  goLeft(steps = 1) {
+    this.coords.x -= steps;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: \
+${this.chipVersion}, Weight: ${this.weight}`;
```

Comment:
```suggestion
    return `Robot: ${this.name}, Chip version:`
      +`${this.chipVersion}, Weight: ${this.weight}`;
```
It is better to break it into two strings